.github/workflows/test-memory-leaks.yml

```diff
@@ -0,0 +1,15 @@
+name: Test for memory leaks
+on: [push]
+jobs:
+  test-memcheck:
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.4.0-preview1 # TODO: Use stable version once 3.4 is out
+          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
+          bundler: latest
+          cache-version: v1 # bump this to invalidate cache
+      - run: sudo apt install -y valgrind && valgrind --version
```

Comment:
You may want to borrow some of the tweaks we've done since to avoid flakiness in installing valgrind: https://github.com/DataDog/dd-trace-rb/blob/master/.github/workflows/test-memory-leaks.yaml#L14